src/Functions/FunctionsStringSearch.h

```diff
@@ -91,23 +91,30 @@ class FunctionsStringSearch : public IFunction
     DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
     {
         if (arguments.size() < 2 || 3 < arguments.size())
-            throw Exception("Number of arguments for function " + getName() + " doesn't match: passed "
-                + toString(arguments.size()) + ", should be 2 or 3.",
-                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+            throw Exception(
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+                "Number of arguments for function {} doesn't match: passed {}, should be 2 or 3.",
```

Comment:
I also prefer not to have a dot in the end of sentence because it can be part of a bigger message, and we will get several punctuation symbols in a row (`.:`)
```
SELECT positionUTF8('1234', 1, 1, 1, 1)

Received exception from server (version 22.11.1):
Code: 42. DB::Exception: Received from localhost:9000. DB::Exception: Number of arguments for function positionUTF8 doesn't match: passed 5, should be 2 or 3.: While processing positionUTF8('1234', 1, 1, 1, 1). (NUMBER_OF_ARGUMENTS_DOESNT_MATCH)
```

```suggestion
                "Number of arguments for function {} doesn't match: passed {}, should be 2 or 3",
```